site/src/components/WorkspaceScheduleForm/WorkspaceScheduleForm.tsx

```diff
@@ -262,6 +278,24 @@ export const WorkspaceScheduleForm: FC<WorkspaceScheduleFormProps> = ({
   )
 }
 
+const ttlShutdownAt = (now: dayjs.Dayjs, workspace: Workspace, tz: string, newTTL: number): string => {
+  if (workspace.latest_build.transition !== "start") {
+    return Language.ttlHelperText
+  }
+  if (newTTL === 0) {
+    return Language.ttlCausesNoShutdownHelperText
+  }
+  const newDeadline = dayjs(workspace.latest_build.updated_at).add(newTTL, "hour")
+  if (newDeadline.isBefore(now)) {
+    return `⚠️ ${Language.ttlCausesShutdownHelperText} ${Language.ttlCausesShutdownImmediately} ⚠️`
+  }
+  if (newDeadline.isBefore(now.add(30, "minute"))) {
+    return `⚠️ ${Language.ttlCausesShutdownHelperText} ${Language.ttlCausesShutdownSoon} ⚠️`
+  }
+  const newDeadlineString = newDeadline.tz(tz).format("hh:mm A z")
+  return `${Language.ttlCausesShutdownHelperText} ${Language.ttlCausesShutdownAt} ${newDeadlineString}.`
+}
```

Comment:
*review:* I'm taking the lazy route and using Unicode characters here instead of icons 🙈

Reply:
Before merge, I'd like to:

- unit test this function
- (I know this goes against every Go-reflex you have): clean up the early-returns to meet our FE conventions

It would look like:

```ts
export const ttlShutdownAt = (now: dayjs.Dayjs, workspace: Workspace, tz: string, newTTL: number): string => {
  if (!isWorkspaceOn(workspace)) {
    return Language.ttlHelperText
  } else if (!newTTL || newTTL <= 0) {
    return Language.ttlCausesNoShutdownHelperText
  } else {
    const newDeadline = dayjs(workspace.latest_build.updated_at).add(newTTL, "hour")

    if (newDeadline.isBefore(now)) {
      return `⚠️ ${Language.ttlCausesShutdownHelperText} ${Language.ttlCausesShutdownImmediately} ⚠️`
    } else if (newDeadline.isBefore(now.add(30, "minutes"))) {
      return `⚠️ ${Language.ttlCausesShutdownHelperText} ${Language.ttlCausesShutdownSoon} ⚠️`
    } else {
      const newDeadlineString = newDeadline.tz(tz).format("hh:mm A z")
      return `${Language.ttlCausesShutdownHelperText} ${Language.ttlCausesShutdownAt} ${newDeadlineString}.`
    }
  }
}
```